editoast/src/views/infra/mod.rs

```diff
@@ -591,17 +602,53 @@ pub mod tests {
             voltage: "0".into(),
             track_ranges: vec![],
         };
+        let rolling_stock: RollingStockModel = get_other_rolling_stock();
+        let rolling_stock_id = rolling_stock.id.unwrap();
+
         let req = create_object_request(infra.id.unwrap(), catenary.into());
         assert_eq!(call_service(&app, req).await.status(), StatusCode::OK);
+        let post_response = call_service(
+            &app,
+            TestRequest::post()
+                .uri("/rolling_stock")
+                .set_json(rolling_stock)
+                .to_request(),
+        )
+        .await;
+        assert_eq!(post_response.status(), StatusCode::OK);
 
         let req = TestRequest::get()
-            .uri(format!("/infra/{}/voltages/", infra.id.unwrap()).as_str())
+            .uri(
+                format!(
+                    "/infra/{}/voltages/?include_rolling_stock_modes=false",
+                    infra.id.unwrap()
+                )
+                .as_str(),
+            )
             .to_request();
         let response = call_service(&app, req).await;
         assert_eq!(response.status(), StatusCode::OK);
+        let voltages: Vec<String> = read_body_json(response).await;
+        assert_eq!(voltages[0], "0");
```

Comment:
I might not have seen it, nor completely understand the model yet, but which assertion is the one verifying that include_rolling_stock_modes is indeed false in the result ? Same question for true.

Reply:
L656 verify that include_rolling_stock_modes is true 
L626 verify that it can get the voltages. I should maybe add assert on the len to show that len(voltages)= 0 when it's false

Reply:
Thanks for the info, looks good to me ;)